### PR TITLE
ENH: Add defaultdictserver tool which serves all requested PVs; spoof-beamline

### PR DESCRIPTION
**Pull request by tacaswell**
:dragon: :smiling_imp: :dragon_face:

### CODE REVIEW

**Comment by klauer** [comment]
This is simply awesome, @tacaswell!

File: scripts/defaultdictserver [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import curio
+from collections import defaultdict
+import caproto.curio.server as ccs
+from caproto import ChannelData
+
+class ReallyDefaultDict(defaultdict):
+    def __contains__(self, key):
+        return True
+
+curio.run(ccs.start_server(ReallyDefaultDict(lambda : ChannelData(value=0)),
```
Comment (danielballan) [inline scripts/defaultdictserver]: whitespace

**Comment by danielballan** [comment]
`spoofbeamline` -> `spoof-beamline`  #namepolice

**Comment by tacaswell** [comment]
using '-' in identifiers just looks *wrong*

File: caproto/curio/server.py [hunk 2]
```diff
@@ -200,6 +200,7 @@ def get_db_entry():
                         self.client_hostname, self.client_username,
                         command.data, command.data_type, command.metadata)
                 except Exception as ex:
+                    logger.exception('in hadle write')
```
Comment (danielballan) [inline caproto/curio/server.py]: This is the first I've seen `logger.exception`. Super useful! Is this supposed to say "invalid write"?

**Pull request merged**